views.py

```diff
@@ -714,6 +714,45 @@ def peak_json():
         result_dict = {'error': {'code': 404, 'message': str(e)}}
     return flask.jsonify(result_dict)
 
+@blueprint.route('/json/mirror/')
+def mirror_json():
+    try:
+        usi1 = flask.request.args.get('usi1')
+        usi2 = flask.request.args.get('usi2')
+
+        plotting_args = _get_plotting_args(flask.request.args, mirror=True)
+        spectrum1, source1, splash_key1 = parsing.parse_usi(usi1)
+        spectrum2, source2, splash_key2 = parsing.parse_usi(usi2)
+        spectrum1, spectrum2 = _prepare_mirror_spectra(spectrum1, spectrum2,
+                                                       plotting_args)
+        similarity, peak_matches = _cosine(
+            spectrum1, spectrum2, plotting_args['fragment_mz_tolerance'],
+            plotting_args['cosine'] == 'shifted')
+
+        spectrum1_dict = {
+            'peaks': _get_peaks(spectrum1),
```

Comment:
Now the peaks are as after spectrum processing (m/z range restriction and base peak normalization). Do we want to return the unmodified spectra instead? That's what happens with the single spectrum JSON API.

Reply:
Fixed.